webapp/graphite/render/functions.py

```diff
@@ -2821,10 +2821,30 @@ def groupByNode(requestContext, seriesList, nodeNum, callback):
     sumSeries(ganglia.by-function.server1.*.cpu.load5),sumSeries(ganglia.by-function.server2.*.cpu.load5),...
 
   """
+  return groupByNodes(requestContext, seriesList, callback, nodeNum)
+
+def groupByNodes(requestContext, seriesList, callback, *nodes):
+  """
+  Takes a serieslist and maps a callback to subgroups within as defined by a common node
+
+  .. code-block:: none
+
+    &target=groupByNodes(ganglia.server*.*.cpu.load*,"sumSeries",1,4)
+
+  Would return multiple series which are each the result of applying the "sumSeries" function
+  to groups joined on the nodes' list (0 indexed) resulting in a list of targets like
+
+  .. code-block :: none
+
+    sumSeries(ganglia.server1.*.cpu.load5),sumSeries(ganglia.server1.*.cpu.load10),sumSeries(ganglia.server1.*.cpu.load15),sumSeries(ganglia.server2.*.cpu.load5),sumSeries(ganglia.server2.*.cpu.load10),sumSeries(ganglia.server2.*.cpu.load15),...
+
+  """
   metaSeries = {}
   keys = []
+  if isinstance(nodes, int):
+    nodes=[nodes]
   for series in seriesList:
-    key = series.name.split(".")[nodeNum]
+    key = '.'.join(series.name.split(".")[n] for n in nodes)
```

Comment:
I'm a bit surprised / confused by how this works. I understand `'.'.join(<somelist>` and i've done list comprehensions before, but it's interesting how the list here is constructed with the `<list>[n]` syntax wherein `[n]` is part of a list comprehension.

```
>>> name = "abc.def.ghi.jkl"
>>> nodes = [0,2]
>>> name.split(".")[n] for n in nodes
  File "<stdin>", line 1
    name.split(".")[n] for n in nodes
                         ^
SyntaxError: invalid syntax
>>> ".".join(name.split(".")[n] for n in nodes)
'abc.ghi'
>>> 
```

anyway, just some python magic / syntactic sugar i guess. it seems to work fine.